workflow/setup_expt.py

```diff
@@ -289,6 +291,15 @@ def edit_baseconfig(host, inputs):
     extend_dict = {"@CCPP_SUITE@": "FV3_GFS_v17_p8", "@IMP_PHYSICS@": 8}
     tmpl_dict = dict(tmpl_dict, **extend_dict)
 
+    # Read in the YAML file to fill out templates and override host defaults
+    yaml_path = inputs.yaml
+    yaml_dict = YAMLFile(path=yaml_path)
+    try:
+        cfg_dict = get_template_dict(yaml_dict['base'])
+    except KeyError:
+        cfg_dict = {}
+    tmpl_dict = dict(tmpl_dict, **cfg_dict)
+
```

Comment:
```suggestion
    try:
        tmpl_dict = dict(tmpl_dict, **get_template_dict(yaml_dict['base']))
    except KeyError:
        pass
```

Reply:
In line 247, can you accept `yaml_dict`?
```python
def edit_baseconfig(host, inputs, yaml_dict):
```

Reply:
Also, move lines 233, 234 before 231 and update 231 to pass `yaml_dict`?
```python
    yaml_path = inputs.yaml
    yaml_dict = YAMLFile(path=yaml_path)

    edit_baseconfig(host, inputs, yaml_dict)
```

Reply:
Ok yeah sure